metals/src/main/scala/scala/meta/internal/worksheets/DecorationWorksheetPublisher.scala

```diff
@@ -40,7 +43,10 @@ class DecorationWorksheetPublisher() extends WorksheetPublisher {
       .map { s =>
         new DecorationOptions(
           s.position().toLsp,
-          new MarkedString("scala", s.details()),
+          new MarkupContent(
```

Comment:
Also, this breaks from the [Decoration Protocol 0.1.0](https://scalameta.org/metals/docs/editors/decoration-protocol.html) with having the hover message being `MarkupContent` rather than `MarkedString`. I can update the docs, however, how do we version the decoration protocol? Do we need to keep history of the v0.1.0, or am I ok to just update it and make it v0.2.0? After we figure this out, I'll also need to update the vs code and coc-metals clients to match this.

This is also causing the integration test suite to fail because we are just looking at the raw string. However, I don't really see this any different than the other hovers, so I think it actually make sense to surround it with ````scala` tics. Thoughts?

Reply:
I sort of just assumed that it was ok to update the protocol. I've gone ahead and updated the decoration protocol docs and updated the changelog just in case someone was using the protocol.

I also updated the worksheet suite to include the new hover text having ```scala tics. I've tested this manually to ensure there is no regression in the way things appear in both VS Code an coc-metals. As far as I can tell, it behaves exactly the same.

Reply:
I think it's fine to do 0.2.0, but probably best if @olafurpg confirms. The only issue I see is that will make the snapshot versions stop working. Maybe we should have the client send the version it is using? And send one or the other (MarkupContent or MarkupString)?

Might be more effort to change it than it's worth, we might need to remove MarkedString at some point, but s it deprecated itself?

Reply:
Talked about his offline, but even with the current version of the extensions, this still works fine. The old shape that was returned was below:

```json
  "hoverMessage": {
        "language": "scala",
        "value": "x: Int \u003d 3"
      }
```
Now, the new format is below:
```json
```json
"hoverMessage": {
        "kind": "markdown",
        "value": "```scala\nx: Int \u003d 3\n```"
      },
```
The current extension is able to handled these both.